src/main/java/org/jnode/fs/xfs/XfsFileSystem.java

```diff
@@ -64,24 +85,36 @@ public final void read() throws FileSystemException {
         }
     }
 
+    public INode getINode(long iNodeNumber) {
+        ByteBuffer allocate = ByteBuffer.allocate(iNodeSize);
+
+        try {
+            this.getApi().read(iNodeSize * iNodeNumber, allocate);
+            return new INode(iNodeNumber, allocate.array() , 0);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading inode: " + iNodeNumber , e);
+        }
+    }
+
     @Override
     public long getTotalSpace() throws IOException {
-        return 0;
+        //TOTAL NUMBER OF BLOCKS..........
```

Comment:
It could be clearer to have this as javadoc, if necessary.

Reply:
change applied